pkg/kubelet/pod_workers.go

```diff
@@ -1645,6 +1645,35 @@ func (p *podWorkers) removeTerminatedWorker(uid types.UID, status *podSyncStatus
 	return true
 }
 
+// killPodNowAsync returns a KillPodFunc that can be used to kill a pod.
+// It is intended to be injected into other modules that need to kill a pod.
+// The kill will happen async, and once it is complete, the lock will be unlocked
+func killPodNowAsync(podWorkers PodWorkers) eviction.KillPodFuncAsync {
```

Comment:
I will probably ask that we not require this change and perhaps allow the pod worker to communicate which pods are being killed (which is part of the intent of https://github.com/kubernetes/kubernetes/pull/125774/commits/99cd22d00b10b3047343f96dc7931241a05fc344 - we want to clearly indicate that).  If you had a way to query the pod worker for the set of pods undergoing eviction (which is the source of truth for pod lifecycle state), that I think wolud simplify the eviction manager and make the change you're trying to accomplish easier.